pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -240,8 +213,17 @@ func (c *Controller) syncOwnedProvisionRequest(ctx context.Context, wl *kueue.Wo
 		attempt := int32(1)
 		shouldCreatePr := false
 		if exists {
-			attempt = getAttempt(ctx, oldPr, wl.Name, checkName)
-			if apimeta.IsStatusConditionTrue(oldPr.Status.Conditions, autoscaling.Failed) {
+			switch {
+			case isCapacityRevoked(oldPr):
```

Comment:
Something is off in terms of separation of concerns.

This controller should be updating the check statuses.

The workload controller then uses the information in the check statuses to decide whether to evict the workload.

That separation allows any admission check to not have to reimplement what to do about `CheckStateRejected`.

Reply:
As we discussed offline it's a bit complexed to unify behavior of the workload controller for every `CheckStateRejected` as it may be caused by different reasons, and it deserves a separate PR.